### PR TITLE
Added Dao and tests for controller and dao

### PR DESCRIPTION
**Pull request by Arashigor**

### CODE REVIEW

File: src/test/java/md/dbalutsel/BookStore/controller/BookStoreControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+        verify(bookService, times(1)).save(book);
+        verifyNoMoreInteractions(bookService);
+    }
+
+    @Test
+    public void deleteBookTest() throws Exception {
+        doNothing().when(bookService).delete(book);
+
+        mockMvc.perform(delete("/books").contentType(APPLICATION_JSON).content(new Gson().toJson(book)))
```
Comment (CornelCaraman) [inline src/test/java/md/dbalutsel/BookStore/controller/BookStoreControllerTest.java]: id

File: src/main/java/md/dbalutsel/BookStore/controller/BookStoreController.java [hunk 2]
```diff
@@ -60,18 +59,8 @@
...
-    @ExceptionHandler(ConstraintViolationException.class)
-    public ResponseEntity<String> exceptionHandler(ConstraintViolationException ex) {
-
-        ConstraintViolationExceptionResponse error =
-                new ConstraintViolationExceptionResponse(ex.getConstraintViolations());
-
-        return new ResponseEntity<>(error.toString(), OK);
+    @DeleteMapping("/books/{id}")
+    public ResponseEntity<?> deleteBook(@PathVariable(name = "id") Integer id) {
+        return (bookService.delete(id)==1) ? new ResponseEntity<>(OK) : new ResponseEntity<>(NOT_FOUND);
```
Comment (CornelCaraman) [inline src/main/java/md/dbalutsel/BookStore/controller/BookStoreController.java]: return no_content insted of ok

File: src/main/java/md/dbalutsel/BookStore/controller/handlers/RestControllerExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        return super.handleExceptionInternal(ex, REST_ERROR_MSG, headers, status, request);
+    }
+
+    @ExceptionHandler(ConstraintViolationException.class)
+    public ResponseEntity<String> constraintViolationExceptionHandler(ConstraintViolationException ex) {
+
+        ConstraintViolationExceptionResponse error =
+                new ConstraintViolationExceptionResponse(ex.getConstraintViolations());
+
+        return new ResponseEntity<>(error.toString(), OK);
```
Comment (CornelCaraman) [inline src/main/java/md/dbalutsel/BookStore/controller/handlers/RestControllerExceptionHandler.java]: why OK if it's a validation errror?

File: src/main/java/md/dbalutsel/BookStore/controller/handlers/RestControllerExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import javax.validation.ConstraintViolationException;
+
+import static md.dbalutsel.BookStore.data.Constants.REST_ERROR_MSG;
+import static org.springframework.http.HttpStatus.OK;
+
+@ControllerAdvice
+public class RestControllerExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @Override
+    protected ResponseEntity<Object> handleExceptionInternal(Exception ex, Object msg, HttpHeaders headers, HttpStatus status, WebRequest request) {
```
Comment (CornelCaraman) [inline src/main/java/md/dbalutsel/BookStore/controller/handlers/RestControllerExceptionHandler.java]: is this needed?

Comment (Arashigor) [reply src/main/java/md/dbalutsel/BookStore/controller/handlers/RestControllerExceptionHandler.java]: it's for global controller exception handler. i didn't make different handler for every possible 4xx or 5xx.

File: src/main/java/md/dbalutsel/BookStore/config/AppConfig.java [hunk 5]
```diff
@@ -1,7 +1,23 @@
...
+import md.dbalutsel.BookStore.service.BookServiceImp;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
 
 @Configuration
+@EnableTransactionManagement
 public class AppConfig {
+    @Bean
+    public BookService bookService() {
```
Comment (CornelCaraman) [inline src/main/java/md/dbalutsel/BookStore/config/AppConfig.java]: anotate BookServiceImp class with @Service and remove this bean definition, the same for BookDaoImpl

File: src/main/java/md/dbalutsel/BookStore/controller/BookStoreController.java [hunk 6]
```diff
@@ -1,50 +1,66 @@
...
+    }
+
+    @GetMapping("/books/{id}")
+    public ResponseEntity<Book> getBookById(@PathVariable(name = "id") Integer id) {
+        Optional<Book> book = bookService.findById(id);
+        return book.map(book1 -> new ResponseEntity<>(book1, OK)).orElseGet(() -> new ResponseEntity<>(NOT_FOUND));
+    }
+
+    @GetMapping(value = "/books", params = "name")
+    public ResponseEntity<Book> getBookByName(@RequestParam("name") String name) {
```
Comment (CornelCaraman) [inline src/main/java/md/dbalutsel/BookStore/controller/BookStoreController.java]: All this methods get book by {...} can be rewritten in a single method

**Pull request merged**